src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionDiagnosticAnalyzer.cs

```diff
@@ -29,29 +31,113 @@ protected override void InitializeWorker(AnalysisContext context)
         {
             var notImplementedExceptionType = context.Compilation.GetTypeByMetadataName(typeof(NotImplementedException).FullName!);
             if (notImplementedExceptionType != null)
-                context.RegisterOperationAction(context => AnalyzeThrow(context, notImplementedExceptionType), OperationKind.Throw);
+            {
+                context.RegisterOperationBlockAction(context =>
+                {
+                    using var _ = SharedPools.Default<ConcurrentSet<Location>>().GetPooledObject(out var reportedLocations);
+                    AnalyzeBlock(context, notImplementedExceptionType, reportedLocations);
+                });
+            }
         });
     }
 
-    private void AnalyzeThrow(OperationAnalysisContext context, INamedTypeSymbol notImplementedExceptionType)
+    private void AnalyzeBlock(
+        OperationBlockAnalysisContext context,
+        INamedTypeSymbol notImplementedExceptionType,
+        ConcurrentSet<Location> reportedLocations)
     {
-        var throwOperation = (IThrowOperation)context.Operation;
-        if (throwOperation is
+        var directThrowsFound = new List<IThrowOperation>();
+
+        // First, collect all direct throw operations that are not inside nested contexts
+        foreach (var block in context.OperationBlocks)
+        {
+            foreach (var operation in block.DescendantsAndSelf())
             {
-                Exception: IConversionOperation
+                if (operation is IThrowOperation
+                    {
+                        Exception: IConversionOperation
+                        {
+                            Operand: IObjectCreationOperation
+                            {
+                                Constructor.ContainingType: INamedTypeSymbol constructedType,
+                            },
+                        },
+                        Syntax: ThrowExpressionSyntax or ThrowStatementSyntax,
+                    } throwOperation &&
+                    notImplementedExceptionType.Equals(constructedType))
                 {
-                    Operand: IObjectCreationOperation
+                    // Only include throws that are directly in the method body
+                    if (IsDirectThrow(throwOperation, context))
```

Comment:
> // Only include throws that are directly in the method body

I don't think this is needed.  It's fine to report *on* the 'throw' wherever it may be.  but we only want to report ont he mthod itself if hte method body is just a direct throw.

Reply:
ok updated to new behavior which always shows on throw, but conditionally on member name